File: Problems/No.1413_Minimum_Value_to_Get_Positive_Step_by_Step_Sum/minimum_value_to_get_positive_step_by_step_sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        
        // parameter
        int min_sum = INT_MAX;

        // start computing
        int sum = 0;
        for (auto &num: nums) {

            sum += num;
            min_sum = std::min(min_sum, sum);

        }

        return (min_sum < 1) ? std::abs(min_sum)+1 : 1;

    }
};

File: Problems/No.1413_Minimum_Value_to_Get_Positive_Step_by_Step_Sum/minimum_value_to_get_positive_step_by_step_sum.py
class Solution(object):
    def minStartValue(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # parameter
        min_sum = float('inf')

        # start computing
        num_sum = 0
        for num in nums:
            num_sum += num
            min_sum = min(min_sum, num_sum)
        
        return 1 if (min_sum > 0) else abs(min_sum)+1
